updater_test.go

```diff
@@ -112,3 +116,49 @@ func newDefaultTestUpdateOptions() UpdateOptions {
 		DestinationPath: filepath.Join(os.TempDir(), "Test"),
 	}
 }
+
+func testServerForUpdateFile(t *testing.T, path string) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		f, err := os.Open(path)
+		assert.NoError(t, err)
+		w.Header().Set("Content-Type", "application/zip")
+		io.Copy(w, f)
+	}))
+}
+
+func testServerForError(t *testing.T, err error) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, err.Error(), 500)
+	}))
+}
+
+func TestUpdater(t *testing.T) {
+	testServer := testServerForUpdateFile(t, testZipPath)
+	defer testServer.Close()
+
+	upr, err := newTestUpdaterWithServer(t, testServer)
+	assert.NoError(t, err)
+	update, err := upr.Update(testUpdateUI{newDefaultTestUpdateOptions()})
+	if assert.NoError(t, err) {
```

Comment:
Hmm, I don't love this branched flow. I prefer to use the `require` package, which just terminates the test in the case that any assertion fails.